pkg/grpc/api_daprinternal.go

```diff
@@ -35,7 +35,8 @@ import (
 
 // CallLocal is used for internal dapr to dapr calls. It is invoked by another Dapr instance with a request to the local app.
 func (a *api) CallLocal(ctx context.Context, in *internalv1pb.InternalInvokeRequest) (*internalv1pb.InternalInvokeResponse, error) {
-	if a.appChannel == nil {
+	appChannel := a.channels.AppChannel()
```

Comment:
Is `a.channels` always guaranteed to be non-empty? Question also applies to other places that make use of it

Reply:
Correct, `channels` will always be non-nil (passed through `Options` or equivalent`), however the underlying channel returned from `AppChannel()` may or may not be nil.

Reply:
We create it as one of the first things we do when creating the `DaprRuntime` https://github.com/dapr/dapr/pull/6762/files#diff-e56f9bb947923c6f8bb4c995bf4b9469a0ed117276223d16b177dcb04399a4bcR183